tests/lambda_endpoints/test_batch_starter.py

```diff
@@ -130,6 +132,74 @@ def sts_client():
         yield boto3.client("sts", region_name="us-west-2")
 
 
+def test_reverse_direction():
+    """Test PreProcessingDependency reverse_direction method."""
+    # Test that they have the same length
+    assert len(downstream_dependents) == len(upstream_dependents)
+    # Check that first downstream dependent is the reverse of the
+    # first upstream dependent
+    first_reversed_dependent = upstream_dependents[0]
+    assert (
+        downstream_dependents[0].relationship == first_reversed_dependent.relationship
+    )
```

Comment:
Does `==` not work on the objects? Then you wouldn't have to compare every attribute.

```suggestion
    assert (
        downstream_dependents[0] == upstream_dependents[0].reverse_direction()
    )
```

Reply:
Yeah, == didn't work. It requires to compare attribute.